scipy/stats/_fit.py

```diff
@@ -1104,6 +1107,7 @@ def rvs(size):
     # Define statistic
     fit_fun = _get_fit_fun(dist, data, guessed_rfd_params, fixed_rfd_params)
     compare_fun = _compare_dict[statistic]
```

Comment:
There is no checking of the value of the argument `statistic` here so it's (currently) possible the key `statistic` not in the dictionary. Usually a good practice is to use the `dict.get(key, default-key-to-get)` method but here you might want to instead raise a `ValueError` (I'm not sure which approach makes the most sense here w/o context).

Reply:
There is checking in the input validation function `_gof_iv` before we get here. Behavior when `statistic` is not among those in the dictionary is tested in `TestGoodnessOfFit::test_gof_iv`.

Reply:
ah I see it now, resolving